horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ActionServiceImpl.java

```diff
@@ -219,6 +223,32 @@ public Action get(int id){
       return ActionMapper.from(action);
    }
 
+   @Override
+   @RolesAllowed(Roles.TESTER)
+   @WithRoles
+   @Transactional
+   public Action update(Action dto) {
+      if (dto.testId <= 0) {
+         throw ServiceException.badRequest("Missing test id");
+      }
+      // just ensure the test exists
+      testService.getTestForUpdate(dto.testId);
+      validate(dto);
+
+      ActionDAO action = ActionMapper.to(dto);
+      if (action.id == null) {
+         action.persist();
+      } else {
+         if (!action.active) {
+            ActionDAO.deleteById(action.id);
+            return null;
+         } else {
+            merge(action);
+         }
+      }
+      em.flush();
```

Comment:
why do we need to flush? can't we let the transaction manager / ORM decide when to flush?

Reply:
The add method had an explicit flush so I added it for update as well, but no good argument except that :)